### ISSUE TITLE
company-gtags backend misbehaving with c preprocessor macros

### ISSUE THREAD

**Issue by aagon** [company-mode/company-mode#1053]
GNU Global can tag C preprocessor macros, either in the form of macro-variables ```#define PAGE_SIZE 4096```, or in the form of macro-functions ```#define PAGE_BELOW(a) ((a / 4096) * 4096)```.

In the second case, we would like the completion to offer ```PAGE_BELOW(a)``` on a prefix like ```PAGE_B```, and not ```PAGE_BELOW(a) ((a / 4096) * 4096)```.

But company-gtags offers the latter, which is not what we want.

This is due to a specific regexp in the function ```company-gtags--annotation``` being present in its greedy version :
```elisp
(defun company-gtags--annotation (arg)
  (let ((meta (get-text-property 0 'meta arg)))
    (when (string-match (concat (regexp-quote arg) "\\((.*)\\).*") meta)
      (match-string 1 meta))))
```
Replacing it with its lazy version solves the problem :
```elisp
(defun company-gtags--annotation (arg)
  (let ((meta (get-text-property 0 'meta arg)))
    (when (string-match (concat (regexp-quote arg) "\\((.*?)\\).*") meta)
      (match-string 1 meta))))
```
This solution keeps intact the completions for regular functions (provided they only have one set of parentheses in their prototype, which is always the case in C/C++). I'll let you decide whether this change is not too violent (it may break other languages supported by GNU Global I am not familiar with). In the meantime, I'll prepare the trivial PR. The changes are below the 15 lines, I should not have to sign anything.

Best,

Aymeric Agon-Rambosson

**Comment by aagon** [comment]
PR #1054.

**Comment by aagon** [comment]
Solved by #1054, closing.
